src/main/java/mods/eln/transparentnode/FuelGenerator.kt

```diff
@@ -0,0 +1,287 @@
+import mods.eln.sixnode.electricalcable.ElectricalCableDescriptor
+import mods.eln.sound.SoundCommand
+import mods.eln.sound.SoundLooper
+import mods.eln.wiki.Data
+import net.minecraft.entity.player.EntityPlayer
+import net.minecraft.item.ItemStack
+import net.minecraft.nbt.NBTTagCompound
+import net.minecraftforge.client.IItemRenderer
+import net.minecraftforge.fluids.FluidContainerRegistry
+import net.minecraftforge.fluids.FluidRegistry
+import org.lwjgl.opengl.GL11
+import java.io.DataInputStream
+import java.io.DataOutputStream
+
+class FuelGeneratorDescriptor(name: String, internal val obj: Obj3D?, internal val cable: ElectricalCableDescriptor,
+                              internal val nominalPower: Double, internal val maxVoltage: Double,
+                              tankCapacityInSecondsAtNominalPower: Double)
+: TransparentNodeDescriptor(name, FuelGeneratorElement::class.java, FuelGeneratorRender::class.java) {
+    companion object {
+        internal fun EfficiencyFactorVsLoadFactor(loadFactor: Double) = when (Utils.limit(loadFactor, 0.0, 1.5)) {
+            in 0.0..0.1 -> 1.375
+            in 0.1..0.2 -> 1.125
+            in 0.2..0.3 -> 1.050
+            in 0.3..0.4 -> 1.025
+            in 0.4..0.5 -> 1.010
+            in 0.5..1.1 -> 1.000
+            in 1.1..1.2 -> 1.050
+            in 1.2..1.3 -> 1.100
+            in 1.3..1.4 -> 1.150
+            in 1.4..1.5 -> 1.5
+            else -> 1.5
+        }
+
+        val TankCapacityInBuckets = 2
+        val GeneratorBailOutVoltageRatio = 0.5
+        val MinimalLoadFractionOfNominalPower = 0.1
+        val VoltageStabilizationGracePeriod = 1.0
+    }
+
+    internal val tankEnergyCapacity = tankCapacityInSecondsAtNominalPower * nominalPower;
+
+    internal val main = obj?.getPart("main")
+    internal val switch = obj?.getPart("switch")
+    internal val cableRenderDescriptor = cable.render
+
+    init {
+        voltageLevelColor = VoltageLevelColor.fromCable(cable)
+    }
+
+    override fun setParent(item: net.minecraft.item.Item, damage: Int) {
+        super.setParent(item, damage)
+        Data.addEnergy(newItemStack())
+    }
+
+    fun draw(on: Boolean = false) {
+        main?.draw()
+        if (on) {
+            UtilsClient.drawLight(switch)
+        } else {
+            switch?.draw()
+        }
+    }
+
+    override fun use2DIcon() = true
+
+    override fun handleRenderType(item: ItemStack, type: IItemRenderer.ItemRenderType) = true
+
+    override fun shouldUseRenderHelper(
+            type: IItemRenderer.ItemRenderType, item: ItemStack,
+            helper: IItemRenderer.ItemRendererHelper) = type != IItemRenderer.ItemRenderType.INVENTORY
+
+    override fun renderItem(type: IItemRenderer.ItemRenderType, item: ItemStack, vararg data: Any) = when(type) {
+        IItemRenderer.ItemRenderType.INVENTORY -> super.renderItem(type, item, *data)
+        else -> {
+            objItemScale(obj)
+            GL11.glPushMatrix()
```

Comment:
Agree, but I do not have this function yet in my branch, so I prefer to change that after the merge (I know am lazy ;-))